src/client/error.rs

```diff
@@ -15,8 +40,30 @@ pub enum Error {
     #[error("Invalid response: {0}")]
     InvalidResponse(String),
 
-    #[error("Server error {0:?} with message \"{1}\"")]
-    ServerError(ProtocolError, String),
+    #[error(
+        "Server error {} with message \"{}\", context: {:?}, payload: {:?}, virtual: {}",
+        protocol_error,
+        string_or_na(error_message),
+        context,
+        payload,
+        is_virtual
+    )]
+    ServerError {
+        /// Protocol-level error message.
+        protocol_error: ProtocolError,
+
+        /// Server message provided by the broker, if any.
+        error_message: Option<String>,
```

Comment:
I wonder if this could be merged into `ServerErrorResponse`?

Reply:
I think most newer Kafka messages now have an error code and a human-readable string (i.e. this message). The response is mostly what else we could extract. So yeah, we could merge it but I guess the resulting enum will rather unhelpful because in many cases we don't have anything else than the error message.